tests/test_external_strategies.py

```diff
@@ -129,3 +134,97 @@ def test_update_energy_measurement_doesnt_call_set_energy_measurement_kWh_for_fu
         ext_strategy_fixture.state.set_energy_measurement_kWh = Mock()
         ext_strategy_fixture.update_energy_measurement()
         ext_strategy_fixture.state.set_energy_measurement_kWh.assert_not_called()
+
+    @pytest.mark.parametrize("ext_strategy_fixture", [LoadHoursExternalStrategy(100),
+                                                      StorageExternalStrategy()],
+                             indirect=True)
+    def test_receive_bid_successful(self, ext_strategy_fixture):
```

Comment:
I'm not sure I understand one thing: these tests are collected under the class `TestPVForecastExternalStrategy`, but they're actually testing other strategies' methods. In this specific case they're not even testing the strategy method, but rather the mixin method.

I would highly recommend to have a one-to-one match between implementation modules and test modules. For example:
- `external_strategies/pv.py` --> `tests/external_strategies/test_pv.py`
- `external_strategies/load.py` --> `tests/external_strategies/test_load.py`
- - `external_strategies/__init__.py` --> `tests/external_strategies/test_init.py` (note: in the future I would refactor the `__init__.py` module and move the code outside of it).

Sorry for not noticing this earlier. Feel free to ignore the comment if you want to just move on with the PR. But I think it's something to keep in mind.

Reply:
Ok, I agree with the proposal to have a one-to-one relationship between implementation and tests. 
in this specific case, it would lead to code duplication (I would have to implement the `test_receive_bid_successful` in both storage and load. I could create a function that could be reused in a `tests/external_strategies/utils.py`module. what do you think?

Reply:
You read my mind about code duplication :D 
One of the reasons for this is that the `bid()` method is duplicated. In this specific case, I would actually vote for keeping the duplication in the tests too, because we're not testing the same method (even though it is identical!).